data/blog/webview2.md

```diff
@@ -0,0 +1,156 @@
+| Node.js                                   | Yes             | No                           |
+| Sandbox                                   | Optional        | Always                       |
+| Requires a Separate Application Framework | No              | Yes                          |
+| Supported Platforms                       | Mac, Win, Linux | Win (Mac on roadmap)         |
+
+## Performance Discussion
+
+Now the big question:
+_Will your app use 50% less memory by switching between one embedding of Chromium and another?_
+Unfortunately not, but that would have been rad.
+
+We created the [scaffolding to compare some simple apps built using Electron, WPF+WebView2, and .NET+WebView2](https://github.com/crossplatform-dev/xplat-challenges/blob/main/results.md). 
+Our goal was to find any egregious differences, rather than deep-dive on micro-benchmarks.
+The source code, and detailed results are available in the [xplat-challenges](https://github.com/crossplatform-dev/xplat-challenges) repository, 
+so you’re free to play around with the frameworks yourself.
+
+Here are the preliminary results.
+Please check the repository for updates.
+
+1. Startup and memory time
+
+    | Technology       | Time |
+    | ---------------- | ---: |
+    | Electron         |   ~4s|
+    | WebView2 + NET5 + WPF |   ~3s|
+    | WebView2 + Win32 (C++)|   ~2s|
+1. Processes Count and Memory
+
+    | Technology | # Processes | Total private bytes | Total working set |
+    | ---        |        ---: |                ---: |              ---: |
+    | Electron   |           4 |             78,940K |          226,396K |
+    | WebView2 (C++)    |           7 |             77,748K |          268,248K |
+    | WebView2 WPF (C#)    |           7 |            102,840K |          307,156K |
+1. Write and Read 10,000 4k Files:
+
+    |          | Write files | Read dir | Sequential Read | Concurrent Read |
+    | ---------|------------:|---------:|----------------:|----------------:|
+    | Electron |     6,218 ms |     21 ms |         8,751 ms |         1,059 ms |
+    | WebView2 (C#) |    10,585 ms |     82 ms |        13,470 ms |         5,936 ms |
+    | WPF w/o WebView2 (C#)     |     4,101 ms |      4 ms |         1,848 ms |           448 ms |
+1. Write and Read 10,000 1MB Files:
+
+    |          | Write files | Read dir | Sequential Read | Concurrent Read |
+    | ---------|------------:|---------:|----------------:|----------------:|
+    | Electron |    29,215 ms |     34 ms |        17,518 ms |         4,602 ms |
+    | WebView2 (C#) |    43,250 ms |     98 ms |        86,222 ms |        35,753 ms |
+    | WPF w/o WebView2 (C#)     |    25,729 ms |      6 ms |        29,536 ms |         3,417 ms |
+1. Sequental IPC Round Trip
+
+    |                              | 1,000 / avg | 10,000 / avg |
+    |------------------------------|------------:|-------------:|
+    | Electron (context isolation) | 211.9 ms / 0.21 ms | 2,400 ms / 0.24 ms |
```

Comment:
Is this via `contextBridge` or via `MessagePort` because one should be faster than the other

Reply:
Ah, answered my own question, if we do this via a `MessagePort` it should be on-pace with the `node integration` one 🤷

Reply:
node integration uses `const { ipcRenderer } = require('electron')`
context isolation uses `contextBridge`

Reply:
> Ah, answered my own question, if we do this via a `MessagePort` it should be on-pace with the `node integration` one 🤷 

Good to know!